org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/scan/DeviceValueMultiPosition.java

```diff
@@ -18,49 +18,38 @@
 
 package org.eclipse.scanning.api.event.scan;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.eclipse.scanning.api.scan.ScannableValueTextProvider;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
- * This is a special Position class to allow both x and y gap values to be returned from a
- * NexusSlitsWrapper scannable. It is a temporary creation until we understand why can't
- * use a List or a Map for a DeviceValue in a DeviceRequest. 
- *
+ * This is a special Position class to allow multiple values to be returned from a scannable. It is a temporary
+ * creation until we understand why can't use a standard List or Map for a DeviceValue in a DeviceRequest. 
  */
-public class DeviceValueMultiPosition {
+public class DeviceValueMultiPosition implements ScannableValueTextProvider {
 	@JsonProperty("values")
-	private List<Double> values = new ArrayList<Double>() {{
-		add(0.0);
-		add(0.0);
-	}};
+	private Map<String, Double> values;
 
-	public double getX_gap() {
-		return values.get(0);
+	public DeviceValueMultiPosition() {
+		values = new LinkedHashMap<String, Double>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)